src/Components/Blazor/Build/src/Microsoft.AspNetCore.Blazor.Build.csproj

```diff
@@ -27,10 +28,26 @@
 
   <ItemGroup>
     <ProjectReference Condition="'$(BuildNodeJS)' != 'false' and '$(BuildingInsideVisualStudio)' != 'true'" Include="$(RepoRoot)src\Components\Web.JS\Microsoft.AspNetCore.Components.Web.JS.npmproj" ReferenceOutputAssembly="false" />
-    <Reference Include="Microsoft.Extensions.CommandLineUtils.Sources" />
+    <Reference Include="Microsoft.Build.Framework" />
+    <Reference Include="Microsoft.Build.Utilities.Core" />
     <Reference Include="System.Reflection.Metadata" />
+    <Reference Include="System.Text.Json" />
   </ItemGroup>
 
+  <Target Name="CopyBuildTask" BeforeTargets="Build" AfterTargets="PrepareForRun">
+    <!-- If you see a build warning here and you mean to update the tasks dll when building in VS, manually stop MSBuild.exe processes. -->
```

Comment:
This is unfortunate, but we reference this task from other samples \ apps in this project. MSBuild locks up the file. This "mitigates" it by copying the task to a different location and using that in the reference-from-source. Failures to copy are ignored.

Reply:
Makes sense.

Reply:
Does this mean that while MSBuild has it locked, it's possible to build this project, but any changes to the codez won't update the running instance of MSBuild?

Reply:
Might be better to let the build fail if it's not possible to update, assuming that only happens in the scenario where you did change something.

Reply:
> assuming that only happens in the scenario where you did change something

Unfortunately that's not true. If you build the sln, which causes the tasks to lock up, and then attempt to "Rebuild", you're borked at that point. I generally don't want users to have to deal with killing the MSBuild process unless they actually happen to modifying the tasks (which should be rare).

Reply:
OK, makes sense. It would still be great to expand the comment in the sources a bit more to capture more of the reasoning here, otherwise I think we're going to keep asking why this works like it does.